pkgs/stdenv/generic/setup.sh

```diff
@@ -1211,7 +1211,7 @@ _defaultUnpack() {
         # We can't preserve hardlinks because they may have been
         # introduced by store optimization, which might break things
         # in the build.
-        cp -pr --reflink=auto -- "$fn" "$destination"
+        cp -r --preserve=mode,timestamps --reflink=auto -- "$fn" "$destination"
```

Comment:
I think this means the permissions of the copy will be determined by the effective umask of the destination. That should not necessarily make it nondeterministic, but it might break a few builds.

Reply:
No, I don't think that's what this does.

From the [GNU coreutils manual section on `cp -p`](https://www.gnu.org/software/coreutils/manual/coreutils.html#index-_002dp-3):

> Using `--preserve` (`-p`) with no attributes listed is equivalent to `--preserve=mode,ownership,timestamps`.

So what this is doing is dropping `cp`'s attempt to copy the ownership information. Note that this would be undone later, since everything in the Nix store is owned by root.

Reply:
Yea, that seems right. I must have misread and thought `mode` was removed instead of `ownership`.